### PR TITLE
Added reports for `Resharper` and `Roslyn` analyzers.

### PR DESCRIPTION
**Pull request by pNovikau**

### CODE REVIEW

**Comment by pNovikau** [comment]
@DiamondDragon please check all TODOs in templates. I am not sure what should be instead of TODO stubs :confused:

File: src/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+| SolutionDirectoryPath | True     | String | Path to the solution folder        |
+| SolutionRelativePath  | True     | String | Relative path to the solution file |
+| Output                | True     | String | Path to the output directory       |
+
+## Example
+
+- Clone [CodeReview.AdHocReview](https://github.com/GodelTech/CodeReview.AdHocReview)
+- Clone [CodeReview.Orchestrator](https://github.com/GodelTech/CodeReview.Orchestrator) (can be any project you want to analyze)
+- Navigate to `CodeReview.AdHockCheck\src` folder
+- Run `.\dotnet-analysis.ps1 -SolutionDirectoryPath C:\CodeReview.Orchestrator -SolutionRelativePath CodeReview.Orchestrator.sln -Output C:\output`
```
Comment (DiamondDragon) [inline src/README.md]: I would suggest to provide examples for Linux and Windows. This is minor improvement but it comes first into my mind when I read this documentation.

File: src/resharper/imports/evaluator-manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  totalFilesCount:
+    query: SELECT COUNT(1) AS count FROM FileDetails
+    type: Scalar
+  totalCodeCount:
+    query: SELECT SUM(Code) FROM FileDetails WHERE Language IN ('C#', 'cshtml')
+    type: Scalar
+  totalLanguagesFilesCount:
+    query: SELECT COUNT(1) as count FROM FileDetails WHERE Language IN ('C#', 'cshtml')
+    ranges:
+      <b><span style='color:red'>(TODO) Number of issues is considered to be high</span></b>:
```
Comment (DiamondDragon) [inline src/resharper/imports/evaluator-manifest.yaml]: What this line is supposed to be used for?

File: src/resharper/imports/evaluator-manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  topTenSevereIssues:
+    query: Select Count(1) as Count, Level, Title, Description from Issues GROUP BY Level, Description, Category ORDER BY Count DESC LIMIT 10
+    type: Collection
+  topTenIssuesByDescription:
+    query: SELECT Description, Count(1) AS count FROM Issues GROUP BY Description ORDER BY Count DESC LIMIT 10
+    type: Collection
+  biggestFileSize:
+    query: SELECT Code FROM FileDetails WHERE Language IN ('C#', 'cshtml') ORDER BY Code DESC LIMIT 1
+    ranges:
+      <b><span style='color:red'>(TODO) File size is considered to be unmaintainable</span></b>:
```
Comment (DiamondDragon) [inline src/resharper/imports/evaluator-manifest.yaml]: Does it make sense to move this HTML into Liquid templtate?

**Review by DiamondDragon — COMMENTED** [review]
PR looks good to me. The only unclear part is template in HTML in evaluation manifest

**Comment by DiamondDragon** [comment]
> @DiamondDragon please check all TODOs in templates. I am not sure what should be instead of TODO stubs 😕

I think we need to define ranges for issues count .E.g. define what number of error is considered to be Ok or no. Here are some suggestions:

Issue count:

* **Good** Total issue count < 100.
* **Average** 100 <= Total issues count < 1000
* **Bad** 1000 <= Total issue count < 1000

File size count:

* **Good** - LOC < 200
* **Avearage** - 200 <= LOC < 2000
* **Too big** - 2000 <= LOC

Some testing is required to adjust these ranges. For now I would suggest to hardcode these values. Later we may change these values as separate PR.

File: src/resharper/imports/evaluator-manifest.yaml (hunk 3 above)
Comment (pNovikau) [reply src/resharper/imports/evaluator-manifest.yaml]: To highlight a bad result. Different colors are set for bad and good results.

File: src/resharper/imports/evaluator-manifest.yaml (hunk 2 above)
Comment (pNovikau) [reply src/resharper/imports/evaluator-manifest.yaml]:
```html
<p class=MsoNormal>
        <b>Total files found:</b> {{ totalLanguagesFilesCount.value }} - {{ totalLanguagesFilesCount.range }}
    </p>
```
https://github.com/GodelTech/CodeReview.AdHocReview/pull/3/files#diff-84859f6b181a95032d0b7259c0429f1fa510a499fe2f403fe9e515f7da77823cR210

**Review by DiamondDragon — APPROVED** [review]

**Pull request merged**